src/lib/atoms/Pill/Pill.tsx

```diff
@@ -0,0 +1,127 @@
+     * When `isFill` prop set to `true`, the `pill` will have a solid fill, providing a more prominent visual appearance. If `false`, the `pill` will be displayed with an outlined style.
+     */
+    isFill?: boolean;
+    /**
+     * Pill size
+     * Possible values: `smallNudge | small | medium`
+     */
+    size?: 'smallNudge' | 'small' | 'medium';
+    /**
+     * Pill icon <br/>
+     * The `Icon` prop accepts a JSX element that will be displayed alongside the `text`
+     */
+    Icon?: JSX.Element;
+    /**
+     * Displays a dot `Icon` when no custom `Icon` is provided. <br/>
+     * If `true`, a dot icon will be shown unless a custom `Icon` is specified. <br/>
+     * If `false`, the dot will be hidden, even if no custom `Icon` is provided.
+     */
+
+    withDot?: boolean;
+    /**
+     * Pill color <br/>
+     * Possible values: `informative | neutral | error | success | warning | purple | lagoon | magenta | slate | inverse`
+     */
+    color?:
+        | 'informative'
+        | 'neutral'
+        | 'error'
+        | 'success'
+        | 'warning'
+        | 'purple'
+        | 'lagoon'
+        | 'magenta'
+        | 'slate'
+        | 'inverse';
+    /**
+     * Additional class for the parent element.
+     * This prop should be used to set placement properties for the element relative to its parent using BEM conventions.
+     */
+    className?: string;
+}
+
+const iconSizes = {
+    smallNudge: 16,
+    small: 20,
+    medium: 20
+} as const;
+
+/**
+ * A Pill component used to display concise information or categorize content. Often used for labels or status indicators, Pill components are visually distinct and can convey different meanings through text and color coding.
+ */
+const Pill: FC<IPillProps> = ({
+    color = 'informative',
+    size = 'medium',
+    text,
+    iconAlignment = 'before',
+    isFill,
+    withDot = true,
+    Icon,
+    className
+}) => {
+    const [isWithDot, setIsWithDot] = useState(withDot);
+    const textRef = useRef<HTMLSpanElement | null>(null);
+    const isTruncated: boolean = useEllipsisDetection(textRef);
+
+    useEffect(() => {
+        setIsWithDot(!Icon && withDot);
+    }, [Icon, withDot]);
+
+    const iconMock = Icon ? (
+        cloneElement(Icon, {
+            size: iconSizes[size],
+            className: 'pill__icon'
+        })
+    ) : isWithDot ? (
+        <Dot size={iconSizes[size]} className="pill__icon" />
+    ) : (
+        ''
+    );
+
+    return (
+        <div
+            className={classNames(`pill pill_size_${size} pill_color_${color}`, className, {
+                [`pill_icon_${isWithDot ? 'before' : iconAlignment}`]: text && iconMock,
+                pill_icon_only: !text,
+                pill_fill: isFill
+            })}
+        >
+            {iconMock}
+            {text && (
+                <Tooltip text={text} isVisible={isTruncated}>
+                    <span ref={textRef} className="pill__text ellipsis-text">
```

Comment:
One more thing, why does `ellipsis-text` permanently exist on the `span` I guess it should be a conditional class and applied to the `span` when it has truncation 🤔